### PR TITLE
Rename `add` processor to `set` processor

### PR DESCRIPTION
**Pull request by martijnvg**
This name makes more sense, because if a field already exists it overwrites it.

### CODE REVIEW

File: docs/plugins/ingest.asciidoc [hunk 1]
```diff
@@ -3,8 +3,8 @@
 
 === Processors
 
-==== Add processor
-Adds one or more fields and associates them with the specified values. If a field already exists,
+==== Set processor
+Sets one or more fields and associates them with the specified values. If a field already exists,
```
Comment (uboness) [inline docs/plugins/ingest.asciidoc]: the code snippet needs an update as well

**Comment by uboness** [comment]
if we remove the `add`, how do we append values to a multi-valued field? One option is to have another processor just for that... another option is to support the following syntax in this `set` processor:

```js
{
  "set" : {
    "field_1" : 42, // default behaviour is to override or add
    "field_2" : { "value" : 42, "append_if_exists" : false } // same as above
    "field_3" : { "value" : 42, "append_if_exists" : true } // this will append the value to the existing one/s
  }
}
```

**Comment by javanna** [comment]
@uboness your comment on arrays makes me think of #14324 which I am working on. Add/set don't really support adding values to an array at the moment. Maybe we can move the discussion to that other issue and do the right thing there?

**Comment by uboness** [comment]
@javanna oh... didn't see that one... will take a look. thx!

File: docs/plugins/ingest.asciidoc (hunk 1 above)
Comment (martijnvg) [inline docs/plugins/ingest.asciidoc]: oops... fixed the snippet

**Comment by javanna** [comment]
LGTM

**Pull request merged**